crates/edr_eth/src/transaction/signed.rs

```diff
@@ -37,6 +37,23 @@ pub enum SignedTransaction {
 }
 
 impl SignedTransaction {
+    /// Returns the effective gas price of the transaction.
+    pub fn effective_gas_price(&self, block_base_fee: U256) -> U256 {
+        match self {
+            SignedTransaction::PreEip155Legacy(tx) => tx.gas_price,
+            SignedTransaction::PostEip155Legacy(tx) => tx.gas_price,
+            SignedTransaction::Eip2930(tx) => tx.gas_price,
+            SignedTransaction::Eip1559(tx) => {
+                block_base_fee
+                    + (tx.max_fee_per_gas - block_base_fee).min(tx.max_priority_fee_per_gas)
+            }
+            SignedTransaction::Eip4844(tx) => {
+                block_base_fee
+                    + (tx.max_fee_per_gas - block_base_fee).min(tx.max_priority_fee_per_gas)
```

Comment:
Nice to have: I'm wondering if we can have a single implementation for the logic in the branches and have some basic unit tests for edge cases?

Reply:
> I'm wondering if we can have a single implementation for the logic in the branches

No, these are two different types: EIP-1559 transaction and EIP-4844 transaction, so it's impossible to merge the two cases

Reply:
> No, these are two different types: EIP-1559 transaction and EIP-4844 transaction, so it's impossible to merge the two cases

A macro could work, but probably an overkill

> Did you have anything in mind?

I'm not sure what would happen if `tx.max_fee_per_gas` is less than `block_base_fee`

(none of these are merge blockers for me)